src/panel/poi/PoiPanel.jsx

```diff
@@ -178,9 +180,32 @@ export default class PoiPanel extends React.Component {
   }
 
   backToList = () => {
+    const { poiFilters } = this.props;
+
     Telemetry.add(Telemetry.POI_BACKTOLIST);
     fire('restore_location');
-    window.app.navigateTo(`/places/?type=${this.props.sourceCategory}`);
+
+    let uri = '/places/';
+
+    // Reconstruct uri
+    for (const filterName of Object.keys(poiFilters)) {
+      if (!poiFilters[filterName]) {
+        continue;
+      }
+
+      uri += uri[uri.length - 1] === '/' ? '?' : '&';
+
+      // TODO use keys as query param ?
+      if (filterName === 'category') {
+        uri += 'type=';
+      } else if (filterName === 'query') {
+        uri += 'q=';
+      }
+
+      uri += poiFilters[filterName];
```

Comment:
By the way, there is a bug here, values which key isn't `category` or `query` will be just concat'd without their key.